src/core/js/tracking.js

```diff
@@ -0,0 +1,84 @@
+define([
+    'core/js/adapt',
+    'core/js/enums/completionStateEnum'
+], function(Adapt, COMPLETION_STATE) {
+
+    var Tracking = Backbone.Controller.extend({
+
+        _config: {
+            _requireContentCompleted: true,
+            _requireAssessmentPassed: false
+        },
+
+        _assessmentState: null,
+
+        initialize: function() {
+            Adapt.once('configModel:loadCourseData', this.loadConfig.bind(this));
+            Adapt.on('app:dataReady', this.setupEventListeners.bind(this));
+        },
+
+        setupEventListeners: function() {
+            // Check if completion requires passing an assessment. 
+            if (this._config._requireAssessmentPassed) {
+                this.listenTo(Adapt, 'assessment:complete', this.onAssessmentComplete);
+            }
+
+            // Check if completion requires completing all content.
+            if (this._config._requireContentCompleted) {
+                this.listenTo(Adapt.course, 'change:_isComplete', this.checkCompletion);
+            }
+        },
+
+        /**
+         * Store the assessment state.
+         * @param {object} assessmentState - The object returned by Adapt.assessment.getState()
+         */
+        onAssessmentComplete: function(assessmentState) {
+            this._assessmentState = assessmentState;
+
+            this.checkCompletion();
+        },
+
+        /**
+         * Evaluate the course and assessment completion.
+         */
+        checkCompletion: function() {
+            if (this._config._requireAssessmentPassed && !this._assessmentState.isComplete) {
```

Comment:
If this function is called because the course is complete, but the assessment hasn't been (odd usecase I know), then you'll get a "cannot find isComplete on undefined" error because the this._assessmentState object won't exist yet:
```js
if (this._config._requireAssessmentPassed && (!this._assessmentState || !this._assessmentState.isComplete)) {
```